dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4897,18 +4897,35 @@ public Node visitWhileLoop(WhileLoopTree tree, Void p) {
 
             // Condition
             addLabelForNextNode(conditionStart);
+
+            // Is condition constant true?
+            boolean isWhileTrue = false;
+
             if (tree.getCondition() != null) {
```

Comment:
I don't think the condition tree could ever be null, could it?
So maybe change this to

```java
if (tree.getCondition() == null) {
  raise error;
}
```

Then you can move the declaration of `isWhileTrue` to line 4906 and simply have `boolean isWhileTrue = condType.isTrue()` and avoid the unnecessary initialization with false.